### ISSUE TITLE
DWAINE scripts can only be run eight times

### ISSUE THREAD

**Issue by penny64** [goonstation/goonstation#5341]
[BUG]

**Describe the bug**
With recent tampering with DWAINE's DEFINEs to combat runtimes has resulted in completely breaking DWAINE scripts, which can now only be run eight times in total per session, which is the only reason it fixed the bug.
**To Reproduce**
1. Create a DWAINE script- or run one of the existing ones, like the ones used for guardbuddies
2. Run it eight times
3. On the ninth try, the command fails with Stack Overflow
4. System must be rebooted to be usable again

**Expected behavior**
DWAINE scripts should be able to be run N times

**Comment by Tarmunora** [comment]
Could not reproduce on local when waiting for scripts to terminate before starting the next

**Comment by Tarmunora** [comment]
After a good 2 hours or so poking at DWAINE with the debugger, I can safely say I have no clue what's going on in there in general. In the specific case, seems that scripts terminate after the a cycle through the `machines` process, and that running multiple scripts concurrently will stack up the `script_iteration` counter, which will stack overflow after reaching 8

**Comment by penny64** [comment]
Specifically I'm just using the scripts for telescience found here
https://wiki.ss13.co/User:Zamujasa/DWAINE_for_Smartasses#Telescience

I use these in my machines so I hit this immediately and I can't even get the offsets

**Comment by penny64** [comment]
![image](https://user-images.githubusercontent.com/13049028/124060803-a3455b80-d9e2-11eb-86fa-d639f4ca8025.png)

On live servers it happens on FROG and TermOS every time, at least from PCs
I'll try a full DWAINE terminal

**Comment by penny64** [comment]
A workaround is a "su" command resets the counter

**Comment by penny64** [comment]
![image](https://user-images.githubusercontent.com/13049028/124062102-046e2e80-d9e5-11eb-82c7-fd4fbcc29f29.png)

Causing a break resets the counter, such as the code

`echo #!|nteleman -p 2 coords $arg0 $arg1 $arg2|nteleman -p 2 scan|nbreak|necho ^ scan`

**Comment by Tarmunora** [comment]
Scripts stick around in the mainframe for a cycle after ending cleanly so that any response can make it back.

**Comment by Tarmunora** [comment]
Append a `break` command to scripts you don't care about receiving a response from and they shouldn't count against concurrent scripts

**Comment by penny64** [comment]
That does work around it but for these telescience ones at least they never end at all and I can't run any scripts without running su or logout

**Comment by Tarmunora** [comment]
Where's the "wontfix" label when you need it 😱

**Comment by frawhst** [comment]
probably fixed by eefcddc ?

**Comment by Tarmunora** [comment]
yes
